presto-delta/src/main/java/com/facebook/presto/delta/DeltaPageSourceProvider.java

```diff
@@ -0,0 +1,402 @@
+import static com.facebook.presto.delta.DeltaTypeUtils.convertPartitionValue;
+import static com.facebook.presto.hive.parquet.HdfsParquetDataSource.buildHdfsParquetDataSource;
+import static com.facebook.presto.hive.parquet.ParquetPageSourceFactory.checkSchemaMatch;
+import static com.facebook.presto.memory.context.AggregatedMemoryContext.newSimpleAggregatedMemoryContext;
+import static com.facebook.presto.parquet.ParquetTypeUtils.columnPathFromSubfield;
+import static com.facebook.presto.parquet.ParquetTypeUtils.getColumnIO;
+import static com.facebook.presto.parquet.ParquetTypeUtils.getDescriptors;
+import static com.facebook.presto.parquet.ParquetTypeUtils.getParquetTypeByName;
+import static com.facebook.presto.parquet.ParquetTypeUtils.getSubfieldType;
+import static com.facebook.presto.parquet.ParquetTypeUtils.lookupColumnByName;
+import static com.facebook.presto.parquet.ParquetTypeUtils.nestedColumnPath;
+import static com.facebook.presto.parquet.predicate.PredicateUtils.buildPredicate;
+import static com.facebook.presto.parquet.predicate.PredicateUtils.predicateMatches;
+import static com.facebook.presto.spi.StandardErrorCode.PERMISSION_DENIED;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toMap;
+import static org.apache.parquet.io.ColumnIOConverter.constructField;
+import static org.apache.parquet.io.ColumnIOConverter.findNestedColumnIO;
+
+public class DeltaPageSourceProvider
+        implements ConnectorPageSourceProvider
+{
+    private final HdfsEnvironment hdfsEnvironment;
+    private final TypeManager typeManager;
+    private final FileFormatDataSourceStats fileFormatDataSourceStats;
+
+    @Inject
+    public DeltaPageSourceProvider(
+            HdfsEnvironment hdfsEnvironment,
+            TypeManager typeManager,
+            FileFormatDataSourceStats fileFormatDataSourceStats)
+    {
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.fileFormatDataSourceStats = requireNonNull(fileFormatDataSourceStats, "fileFormatDataSourceStats is null");
+    }
+
+    @Override
+    public ConnectorPageSource createPageSource(
+            ConnectorTransactionHandle transactionHandle,
+            ConnectorSession session, ConnectorSplit split,
+            ConnectorTableLayoutHandle layout,
+            List<ColumnHandle> columns,
+            SplitContext splitContext)
+    {
+        DeltaSplit deltaSplit = (DeltaSplit) split;
+        DeltaTableLayoutHandle deltaTableLayoutHandle = (DeltaTableLayoutHandle) layout;
+        DeltaTableHandle deltaTableHandle = deltaTableLayoutHandle.getTable();
+
+        HdfsContext hdfsContext = new HdfsContext(session);
```

Comment:
NIT: HdfsContext(ConnectorSession) is deprecated and it is preferable to use HdfsContext(ConnectorSession session, String schemaName, String tableName, String tablePath, boolean isNewTable) wherever possible.